abr_control/controllers/path_planners/path_planner.py

```diff
@@ -1,16 +1,459 @@
+            )
+
+        self.position_path = []
+
+        # the distance covered over time with respect to our velocity profile
+        path_steps = np.cumsum(self.stacked_vel_profile * self.dt)
+
+        # step along our curve, with our next path step being the distance
+        # determined by our velocity profile, in the direction of the path curve
+        for ii in range(0, len(self.stacked_vel_profile)):
+            shiftx = XYZ[0](path_steps[ii])
+            shifty = XYZ[1](path_steps[ii])
+            shiftz = XYZ[2](path_steps[ii])
+            shift = np.array([shiftx, shifty, shiftz])
+            self.position_path.append(shift)
+
+        self.position_path = np.asarray(self.position_path)
+
+        # get our 3D vel profile components by differentiating the position path
+        # our velocity profile is 1D, to get the 3 velocity components we can
+        # just differentiate the position path. Since the distance between steps
+        # was determined with our velocity profile, we should still maintain the
+        # desired velocities. Note this may break down with complex, high
+        # frequency paths
+        self.velocity_path = np.asarray(
+            np.gradient(self.position_path, self.dt, axis=0)
+        )
+
+        # check if we received start and target orientations
+        if isinstance(start_orientation, (list, (np.ndarray, np.generic), tuple)):
+            if isinstance(target_orientation, (list, (np.ndarray, np.generic), tuple)):
+                # Generate the orientation portion of our trajectory.
+                # We will use quaternions and SLERP for filtering orientation path
+                quat0 = transform.quaternion_from_euler(
+                    start_orientation[0],
+                    start_orientation[1],
+                    start_orientation[2],
+                    axes=self.axes,
+                )
+
+                quat1 = transform.quaternion_from_euler(
+                    target_orientation[0],
+                    target_orientation[1],
+                    target_orientation[2],
+                    axes=self.axes,
+                )
+
+                self.orientation_path = self.OrientationPlanner.match_position_path(
+                    orientation=quat0,
+                    target_orientation=quat1,
+                    position_path=self.position_path,
+                )
+                if self.verbose:
+                    self.log.append(
+                        f"{c.blue}start_orientation={start_orientation} | "
+                        + f"target_orientation={target_orientation}{c.endc}"
+                    )
+            else:
+                raise NotImplementedError(
+                    f"{c.red}A target orientation is required to generate path{c.endc}"
+                )
+
+            self.orientation_path = np.asarray(self.orientation_path)
+            # TODO should this be included? look at proper derivation here...
```

Comment:
it's not clear to me here what's being asked. should the angular velocities be included?

Reply:
Sorry meant to remove that, I checked some other links and it looks like angular velocity is just the rate of change of orientation. 
Check out the first equation under `angular velocity`
https://courses.lumenlearning.com/physics/chapter/6-1-rotation-angle-and-angular-velocity/

Reply:
should i remove both the TODO and the hyperlink?